File: vhosts/blog/nuldoc-src/render.ts
import { Node } from "./dom.ts";
import { renderHtml } from "./renderers/html.ts";

// export type RendererType = "html" | "xml";
export type RendererType = "html";

export function render(root: Node, renderer: RendererType): string {
  if (renderer === "html") {
    return renderHtml(root);
  } else {
    return renderHtml(root);
  }
}

File: services/nuldoc/lib/nuldoc/render.rb
module Nuldoc
  class Renderer
    def render(root, renderer_type)
      case renderer_type
      when :html
        HtmlRenderer.new.render(root)
      when :xml
        XmlRenderer.new.render(root)
      else
        raise "Unknown renderer: #{renderer_type}"
      end
    end
  end
end
